core/src/main/java/io/grpc/internal/GrpcUtil.java

```diff
@@ -618,4 +624,20 @@ private static String getImplementationVersion() {
     }
     return "";
   }
+
+  /**
+   * Strips away any reserved headers for client requests.
+   */
+  public static void discardHttp2RequestHeaders(Metadata headers) {
```

Comment:
I'm not excited about how far away from the usages these are. OkHttp in particular would be very easy to put the discards near where the other keys are added. With Netty, the filtering could happen in GrpcHttp2OutboundHeaders.